File: mykng/kb-auth/src/main/java/com/kb/auth/entity/RequestLog.java
package com.kb.auth.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("sys_request_log")
public class RequestLog {
    @TableId(type = IdType.AUTO)
    private Long id;

    private String traceId;

    private Long userId;

    private String username;

    private String httpMethod;

    private String requestUri;

    private String controllerMethod;

    private String requestArgs;

    private String responseResult;

    private Long costMs;

    private String status;

    private String exception;

    private String ip;

    private String userAgent;

    private String serviceName;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
}

File: mykng/kb-web/src/api/requestLog.ts
import request from './index'
import type { R, PageResult, PageParams } from '@/types'

export interface RequestLog {
  id: number
  traceId?: string
  userId?: number
  username?: string
  httpMethod?: string
  requestUri?: string
  controllerMethod?: string
  requestArgs?: string
  responseResult?: string
  costMs?: number
  status?: string
  exception?: string
  ip?: string
  userAgent?: string
  serviceName?: string
  createdAt: string
}

export function getRequestLogList(params: PageParams & {
  traceId?: string
  userId?: number
  httpMethod?: string
  uri?: string
  status?: string
  serviceName?: string
  startTime?: string
  endTime?: string
}) {
  return request.get<R<PageResult<RequestLog>>>('/auth/request-log/list', { params })
}

export function getRequestLogDetail(id: number) {
  return request.get<R<RequestLog>>(`/auth/request-log/${id}`)
}
